File: evidence/generated-systems/angular/governance-on-angular/src/app/models/PrivacyNotice.js

// Define collection and schema for PrivacyNotice
export interface PrivacyNotice {
    title:
	type : string
    audience:
	type : string
    versionLabel:
	type : string
    publicationDate:
	type : Date
    publicationUrl:
	type : URL
    ProcessingActivities:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataProcessingActivity' }]
    Organization:
	type : Schema.Types.ObjectId
    Consents:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Consent' }]
    Status:
 	type : String
#
    collection: 'privacyNotices'
}

File: evidence/generated-systems/django/governance-on-django/governanceOnDjango/models/PrivacyNotice.py
from django.db import models
from governanceOnDjango.models.DocumentStatus import DocumentStatus

#======================================================================
# 
# Encapsulates data for model PrivacyNotice
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class PrivacyNotice Declaration
#======================================================================
class PrivacyNotice (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	title = models.CharField(max_length=200, null=True)
	audience = models.CharField(max_length=200, null=True)
	versionLabel = models.CharField(max_length=200, null=True)
	publicationDate = models.DateField(null=True)
	publicationUrl = URL
	processingActivities = models.ManyToManyField('DataProcessingActivity',  blank=True, related_name='+')
	organization = models.ForeignKey('Organization', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	consents = models.ManyToManyField('Consent',  blank=True, related_name='+')
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in DocumentStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.title
		str = str + self.audience
		str = str + self.versionLabel
		str = str + self.publicationDate
		str = str + self.publicationUrl
		str = str + self.status
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "PrivacyNotice";
    
	def objectType(self):
		return "PrivacyNotice";
